lib/compat/wordpress-6.9/rest-api.php

```diff
@@ -0,0 +1,31 @@
+<?php
+/**
+ * PHP and WordPress configuration compatibility functions for the Gutenberg
+ * editor plugin changes related to REST API.
+ *
+ * @package gutenberg
+ */
+
+/**
+ * Adds export theme link relation to the block theme responses.
+ *
+ * @param WP_REST_Response $response The response object.
+ * @param WP_Theme         $theme    Theme object used to create response.
+ * @return WP_REST_Response Modified response object.
+ */
+function gutenberg_rest_theme_export_link_rel( $response, $theme ) {
+	if ( ! empty( $response->get_links() ) && $theme->is_block_theme() ) {
+		$response->add_link(
+			'https://api.w.org/export-theme',
+			rest_url( 'wp-block-editor/v1/export' ),
+			array(
+				'targetHints' => array(
+					'allow' => current_user_can( 'export' ) ? array( 'GET' ) : array(),
+				),
```

Comment:
Ideally, this should use the same method as `self` links, but I went with [KISS](https://en.wikipedia.org/wiki/KISS_principle) for now 😄 Maybe in the future we can introduce a flag to opt-in `targetHints` generation for any rel link.

@TimothyBJacobs, what do you think?

https://github.com/WordPress/gutenberg/blob/accde16096464ac832fbe8e249a3b61535800800/lib/compat/wordpress-6.7/class-gutenberg-rest-server.php#L86-L90